### PR TITLE
Add DB2 documentation to user manual

### PR DESCRIPTION
**Pull request by dwadler**
Hopefully everything is correct in the branch db2-doc. 

I somehow ended up with branches DB2-Doc and db2-doc and DB2-Doc didn't get my latest commits when "pushed" from a Windows environment. The pull request for DB2-Doc has been deleted.

### CODE REVIEW

File: scripts/subst.py [hunk 1]
```diff
@@ -340,6 +341,7 @@
 |mIconColorSwatches|
 |mIconColorWheel|
 |mIconDataDefine|
+|mIconDb2|
```
Comment (DelazJ) [inline scripts/subst.py]: @rduivenvoorde do we need to (manually) update this file? Have never done it.
I also wonder what it's about.

File: source/conf.py [hunk 2]
```diff
@@ -528,6 +528,8 @@
    :width: 1.5em
 .. |formAnnotation| image:: /static/common/mActionFormAnnotation.png
    :width: 1.5em
+.. |mActionAddDb2Layer| image:: /static/common/mActionAddDb2Layer.png
+   :width: 1.5em
```
Comment (DelazJ) [inline source/conf.py]: Better use |addDb2Layer| as substitution. We've got rid of mAction, mIcon prefixes... from substitutions
And keep it sorted

File: source/conf.py [hunk 3]
```diff
@@ -909,6 +911,8 @@
    :width: 1.5em
 .. |processingToolbox| image:: /static/common/alg.png
    :width: 1.5em
+.. |mIconDb2| image:: /static/common/mIconDb2.png
+   :width: 1.5em
```
Comment (DelazJ) [inline source/conf.py]: as above, simply |db2|

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 4]
```diff
@@ -879,3 +879,157 @@ To load a layer from Oracle Spatial, perform the following steps:
...
+The **Authentication** and **Configurations** tabs in the DB2 Connection
+dialog allow you to provide user name and password information that can be used
+to test the connection and that can be saved for use in a new QGIS session.
+
+If you specify the saving of user name and password under the **Authentication**
+tab, this information will be stored in plain text which may have security concerns.  
+  
+The **Configurations** tab allows you to create a `configuration` which saves
+the user name and password in encrypted form so that it is not visible.
+You need to first specify a `master` password which will be used to decrypt
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: ``master`` ? If i'm not wrong, using a single ` character will not apply a special rendering to the text, in html.

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 5]
```diff
@@ -879,3 +879,157 @@ To load a layer from Oracle Spatial, perform the following steps:
...
+the saved information. This password is specific to the particular QGIS installation
+and is not retrievable if lost.
+You then specify a name to associate this `configuration` with a particular
+database connection.
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Are the configurations and authentication matters above related to the authentication system of QGIS (http://docs.qgis.org/testing/en/docs/user_manual/auth_system/index.html)? In which case, would that be possible to add a link to the main chapter?

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 6]
```diff
@@ -879,3 +879,157 @@ To load a layer from Oracle Spatial, perform the following steps:
...
+   which features to load from the layer or use the **[Build query]** button
+   to start the :guilabel:`Query builder` dialog.
+*  Find the layer(s) you wish to add in the list of available layers.
+*  Select it by clicking on it. You can select multiple layers by holding
+   down the :kbd:`Shift` key while clicking. See section
+   :ref:`vector_query_builder` for information on using the DB2
+   Query Builder to further define the layer.
+*  Click on the **[Add]** button to add the layer to the map.
+
+Or more simply, expand the |mIconDb2| :menuselection:`DB2` connection in the browser tree
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Do you mean the QGIS Browser panel?

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 7]
```diff
@@ -879,3 +879,157 @@ To load a layer from Oracle Spatial, perform the following steps:
...
+and double-click the name of the layer.
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: if QGIS Browser then drag-and-drop might also work (to add layer to canvas)

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 8]
```diff
@@ -879,3 +879,157 @@ To load a layer from Oracle Spatial, perform the following steps:
...
+
+In order to work effectively with DB2 spatial tables in QGIS, it is important that
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Could this be exposed as a tip, note or warning? I don't know what better suits... just an idea

File: source/conf.py (hunk 2 above)
Comment (dwadler) [inline source/conf.py]: When I added the DB2 documentation last March, I was following the conventions that I found in the source at the time. I will make changes as suggested.

**Comment by DelazJ** [comment]
Added some comments. I don't have db2 connection so those were all about doc writing rather than the description itself.
Thanks for contributing

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 5 above)
Comment (dwadler) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: It was a challenge figuring out the authentication as I couldn't find any documentation of it anywhere. If it now exists, I will put in a link and maybe remove this section.

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 6 above)
Comment (dwadler) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: No, I mean the file system tree, database, WMS, etc panel on the left side of QGIS Desktop window..

File: source/conf.py (hunk 2 above)
Comment (DelazJ) [inline source/conf.py]: Yes you are right. That has been decided later. Thanks

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 5 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Yes I can imagine the challenge it was. I never tried to dig into that feature, waiting for a doc. And We do have a documentation about this feature now.

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 6 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Different from this (http://docs.qgis.org/testing/en/docs/user_manual/qgis_browser/qgis_browser.html#qgis-browser-widget)?

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 4 above)
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Could you add a reference to the authentication chapter?

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 6 above)
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: @dwadler in QGIS this panel is called QGIS Browser panel :)

**Comment by dwadler** [comment]
All of the reviewer comments have been addressed and hyperlinks verified.

Are we ready to merge into Master?

**Comment by yjacolin** [comment]
done. Thanks!

**Pull request merged**